services/web/src/components/form-fields/Address.js

```diff
@@ -0,0 +1,287 @@
+import { get, set } from 'lodash';
+import React from 'react';
+import { Form, Segment, Dropdown } from 'semantic';
+import { loadScript } from 'utils/script';
+import CountriesField from 'components/form-fields/Countries';
+import UsStates from './UsStates';
+import { GOOGLE_API_KEY } from 'utils/env';
+import { Loader } from 'semantic-ui-react';
+
+function extractGoogleAddressComponent(
+  addressComponents,
+  type,
+  attribute = 'long_name'
+) {
+  const result = addressComponents.filter((c) => c.types.includes(type))[0];
+  return result ? result[attribute] : '';
+}
+
+function composeDefaultLookupValue(address) {
+  if (!address || !address.line1) return '';
+  const components = [address.line1];
+  if (address.city) {
+    components.push(address.city);
+  }
+  if (address.stateOrProvince) {
+    components.push(address.stateOrProvince);
+  }
+  if (address.countryCode) {
+    components.push(address.countryCode);
+  }
+  return components.join(', ');
+}
+
+export default class Address extends React.Component {
+  state = {
+    initializing: true,
+    manualEntry: false,
+    error: false,
+    placeOptions: [],
+    lookupValue: composeDefaultLookupValue(this.props.value),
+  };
+
+  componentDidMount() {
+    this.loadGoogleDependency();
+  }
+
+  loadGoogleDependency() {
+    const { disableLookup } = this.props;
+    if (window.google) {
+      this.initializeLookup();
+      this.setState({ initializing: false });
+      return;
+    }
+    if (disableLookup || !GOOGLE_API_KEY) {
+      this.setState({ initializing: false, manualEntry: true });
+      return;
+    }
+    loadScript(
+      `https://maps.googleapis.com/maps/api/js?key=${GOOGLE_API_KEY}&libraries=places`
+    );
+    this.loadingInterval = setInterval(() => {
+      if (window.google) {
+        clearInterval(this.loadingInterval);
+        this.initializeLookup();
+        this.setState({ initializing: false });
+      }
+    }, 200);
+  }
+
+  initializeLookup() {
+    this.autocompleteService = new window.google.maps.places.AutocompleteService();
+    this.geocoderService = new window.google.maps.Geocoder();
+  }
+
+  setField = (e, { name, value }) => {
+    const currentValue = this.props.value || {};
+    set(currentValue, name, value);
+    this.props.onChange(e, { name: this.props.name, value: currentValue });
```

Comment:
Seems weird to me
E.g this is suppose to be single "<Form.Field>" but it's actually multiple fields.


I would expect the onChange to return the full object.

```
onChange({ value: { ... fulladdressobject } })
```

This would make it more like a single field. Possible it shouldn't even trigger onChange unless it got a complete address to emit.

Reply:
I'm making it behave exactly like the other Semantic UI input components.

Reply:
I like that personally everything in form-fields should speak the same language

Reply:
Yeah the event param is a bit useless. Some day it would be good to wrap it all (and also align `checked` and `value`!!)

Reply:
i got no problem what it speak the same language, thats good. 

This is what you expect
```
<Input 
 value={this.state.value} 
 onChange={(e, { value }) => this.setState({ value }}
/>
```
Thats all good simple (the above)

Now if we do this for the address component
```
<Address 
 value={this.state.value} 
 onChange={(e, { value }) => this.setState({ value }}
/>
```
This wont work. The onChange is only for the given address field you change (e.g. only a partial value) 

e.g if value is 
```
const value =  {
   country: "NL"
   street: "soemthing",
...
}
```

and you change the address.country field to "US" the expect onChange event would be 

```
{
   country: "US"
   street: "soemthing",
...
}
```
instead you get 

just { value: "US" }